akka-docs/src/main/paradox/typed/persistence-lifecycle.md

```diff
@@ -0,0 +1,91 @@
+ 
+## Recovery
+
+It is strongly discouraged to perform side effects in `applyEvent`,
+Side effects should be performed once recovery has completed as a reaction to the `RecoveryCompleted` signal @scala[`receiveSignal` handler] @java[by overriding `receiveSignal`]
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #recovery }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #recovery }
+
+The `RecoveryCompleted` contains the current `State`.
+
+@ref[Snapshots)[persistence-snapshot.md] can be used for optimizing recovery times.
+
+## Retention - snapshots and events
+
+Retention of snapshots and events are controlled by a few factors. Deletes to free up space is currently available.
+
+@@@ note
+
+The retention API itself [is under current development, e.g. #26545](https://github.com/akka/akka/issues/26545)
+ 
+@@@ note
+  
+### Snapshot deletion
+
+To free up space, an event sourced actor can automatically delete older snapshots 
+if there are saved snapshots which match the `SnapshotSelectionCriteria`. 
+Deletion of snapshots is currently based on a configured or default `RetentionCriteria` 
+from `withRetention`combined with either the `snapshotWhen` or `snapshotEvery` methods. 
+If snapshots are enabled, the deletion sliding window leverages the `RetentionCriteria` to construct
+a `SnapshotSelectionCriteria` based on the latest sequence number. 
+ 
+@@@ note
+
+[This API will be simplified soon](https://github.com/akka/akka/issues/26544).
+
+@@@ note
+  
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshotDeletes }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshotDeletes }
+
+On async deletion, either a `SnapshotCompleted` or `SnapshotFailed` is returned. Successful completion is logged by the system at log level `debug`, failures at log level `error`.
+You can leverage `EventSourcedSignal` to react to outcomes with `EventSourcedBehavior.receiveSignal`.
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #fullDeletesSampleWithSignals }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshotDeletesSifullDeletesSampleWithSignalsgnal }
+
+## Event deletion
+If snapshot-based recovery is enabled, you can elect to have the system automatically delete older events by enabling `RetentionCriteria.deleteEventsOnSnapshot`, which is disabled by default. 
```

Comment:
Maybe mention that it is up to the journal implementation if events are actually really removed from storage, and also that deleting events prevents future replaying of old events to apply new state etc.

Reply:
maybe incorporate some of this: https://doc.akka.io/docs/akka/current/persistence.html?language=scala#message-deletion